convert2rhel/main.py

```diff
@@ -66,6 +66,9 @@ def main():
         loggerinst.task("Prepare: Determine RHEL variant")
         rhelvariant.determine_rhel_variant()
 
+        # prohibit converting if system is not in single-user mode
+        utils.require_single_user_mode()
```

Comment:
I'd move this up, right below calling the `utils.require_root()`. Both functions seem similar in purpose to me, so I feel they should be closer to each other in the code.
@ZhukovGreen, @SpyTec, WDYT?

Reply:
Sounds logical to me @bocekm . However, error handling (starts on line 104) won't work. I think this is OK.

Reply:
@bocekm What is the reason for that? My reasoning for moving it down was so that the user has gone through EULA and know whether the system can be converted.

If we move it further up they might end up rebooting to single-user only to then not go through with the conversion due to incompatible system or not agreeing to EULA

Reply:
@SpyTec, you're right, that makes sense.
@fellipeh, please update the comment to include a comment like:
_We check for the single-user mode only after the resolve_system_info() call because that function stops the conversion in case the system vendor/major version is not supported and we don't want users to go through booting into single-user mode to just find this out._